### PR TITLE
Fix #30

### LINKED ISSUE #30: Which distributions can be used in rsad simulations?
I couldn't find in any of the reasonable places (man page, vignettes, even the function source code) an easy way to figure out for which sads we can run "rsad". Both "lnorm" and "geom" work, but logseries and broken stick do not:

```R
> rsad(S=100, frac=0.1, sad="bs")
```
> Error in do.call(sad, c(list(n = S), dots)) : 
>   could not find function "rbs"

```R
> rsad(S=100, frac=0.1, sad="ls")
```
> Error in do.call(sad, c(list(n = S), dots)) : 
>   could not find function "rls"

### PR DESCRIPTION
**Pull request by andrechalom**
using match.arg to validate input, fixes issue #30

### CODE REVIEW

File: R/rsad.R [hunk 1]
```diff
@@ -1,5 +1,6 @@
-rsad <- function(S,frac,sad, Pois.samp=TRUE,k,zeroes=FALSE,ssize=1, ...){
+rsad <- function(S, frac, sad=c("gamma", "geom", "lnorm","rnbinom", "rpoilog", "rweibull"), Pois.samp=TRUE, k, zeroes=FALSE, ssize=1, ...){
```
Comment (piklprado) [inline R/rsad.R]: Why not 
rsad <- function(S, frac, sad=c("gamma", "geom", "lnorm", "nbinom", "poilog", "weibull")
?

Comment (piklprado) [inline R/rsad.R]: Why not
rsad <- function(S, frac, sad=c("gamma", "geom", "lnorm", "nbinom", "poilog", "weibull")
?

Comment (piklprado) [inline R/rsad.R]: Why not
rsad <- function(S, frac, sad=c("gamma", "geom", "lnorm", "nbinom", "poilog", "weibull")
?

**Comment by piklprado** [comment]
Some issues about changes in rsad. First one:
 - I didn't get why include the prefix 'r' for some distributions in the argument 'sad' in line 1 of rsad:
`rsad <- function(S, frac, sad=c("gamma", "geom", "lnorm", "rnbinom", "rpoilog", "rweibull"), ...`
since line 4 add an 'r' to the distribution root name provided in this argument
`sad <- paste("r",sad,sep="")`

**Comment by andrechalom** [comment]
Yes, the accepted values in this function should be ..., "nbinom", "poilog", "weibull" instead of ..., "rnbinom", "rpoilog", "rweibull". Fixed.

File: R/rsad.R [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
     if(ssize<1)stop("ssize must be at least one")
+	sad <- match.arg(sad)
     sad <- paste("r",sad,sep="")
```
Comment (piklprado) [inline R/rsad.R]: Might be also:

`sad <- get ( paste("r", sad, sep=""), mode = "function" )`

Comment (andrechalom) [inline R/rsad.R]: Fixed.

**Comment by piklprado** [comment]
I do not think that the  `sad` argument in `rsad` needs a `match.arg` checking. The idea is that any distribution with random number generation  could be used. The distribution root name is provided by this argument and then line 

`sad <- paste("r",sad,sep="")`

Creates a name of the corresponding random generation fuction that is the passed to `do.call`. If there is not a random gereneration function for the given root name this will return an error, which seems ok to me. It is up to the user to provide a distribution that has a random generation function. 

Changing the line above to:

`sad <- get ( paste("r", sad, sep=""), mode = "function" )`

May return a more informative error message.

**Comment by andrechalom** [comment]
OK. Including that info in the man page and updating the merge request.

**Pull request merged**